.github/workflows/checks.yml

```diff
@@ -0,0 +1,48 @@
+# This workflow will install Python dependencies, run tests and lint both for the action server and core
+
+name: Python Checks
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  check-core:
```

Comment:
Since the workflow is already named "Python Checks" maybe we could simply name the jobs `core` and `action-server`